pkg/profiler/profiler.go

```diff
@@ -712,22 +713,24 @@ func (p *CgroupProfiler) sendProfile(ctx context.Context, prof *profile.Profile)
 		return err
 	}
 
-	var labeloldformat []*profilestorepb.Label
-
+	var (
+		labelOldFormat = make([]*profilestorepb.Label, 0, len(p.Labels()))
+		i              = 0
+	)
 	for key, value := range p.Labels() {
-		labeloldformat = append(labeloldformat,
-			&profilestorepb.Label{
-				Name:  string(key),
-				Value: string(value),
-			})
+		labelOldFormat[i] = &profilestorepb.Label{
+			Name:  string(key),
+			Value: string(value),
+		}
+		i++
 	}
```

Comment:
Nothing I want changed, but curious, why is this a better way? :thinking:

Reply:
Rather than appending to a dynamically sized slice, this pre-allocates and fixes the slice size, then writing the value in specific positions. Just to avoid growing the slice. Really a micro optimization but static analyzer was smart enough to catch and point it.